core/src/main/scala/com/pingcap/tispark/write/TiDBWriter.scala

```diff
@@ -32,27 +34,27 @@ object TiDBWriter {
     options.checkWriteRequired()
     TiExtensions.getTiContext(sparkSession) match {
       case Some(tiContext) =>
-        val conn = TiDBUtils.createConnectionFactory(options.url)()
-
-        try {
-          val tableExists = TiDBUtils.tableExists(conn, options)
-          if (tableExists) {
-            saveMode match {
-              case SaveMode.Append =>
-                TiBatchWrite.write(df, tiContext, options)
-
-              case _ =>
-                throw new TiBatchWriteException(
-                  s"SaveMode: $saveMode is not supported. TiSpark only support SaveMode.Append.")
-            }
-          } else {
-            throw new TiBatchWriteException(
-              s"table `${options.database}`.`${options.table}` does not exists!")
-            // TiDBUtils.createTable(conn, df, options, tiContext)
-            // TiDBUtils.saveTable(tiContext, df, Some(df.schema), options)
+        var tableExists: Boolean = false
+        val tiSession: TiSession = tiContext.tiSession
+        val catalog: Catalog = tiSession.getCatalog
+        val table: TiTableInfo = catalog.getTable(options.database, options.table)
+        if (table != null)
+          tableExists = true
```

Comment:
```suggestion
          tableExists = true
```
val tableExists = (table!=null)